src/clib/initialize_rates.cpp

```diff
@@ -250,6 +250,58 @@ int add_h2dust_S_reaction_rate(double **rate_ptr, double units, chemistry_data *
     return GR_SUCCESS;
 }
 
+/// sets up the species-specific h2dust grain coefficient grids
+int setup_h2dust_grain_rates(chemistry_data* my_chemistry,
+                             chemistry_data_storage *my_rates,
+                             double kUnit) {
+
+  //H2 formation on dust grains with C and S compositions
+  if (
+    (add_h2dust_C_reaction_rate(&my_rates->h2dustC, kUnit, my_chemistry)
+      != GR_SUCCESS) ||
+    (add_h2dust_S_reaction_rate(&my_rates->h2dustS, kUnit, my_chemistry)
```

Comment:
Maybe now is the time to right this wrong convert S to Si to denote this being silicon.